feat(curriculum): Create a JavaScript Math Objects and Operators Quiz (#56539)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/quiz-javascript-math/66edc3ab8c6413c344f401bf.md

#### --text--

> What is the result of this code:
> const num1 = Number('888');
> const num2 = new Number('777');
> 
> console.log(num1 == 888); //statement 1
> console.log(num2 == 777); // statement 2

Reviewer: ````suggestion
What is the result of this code:

```js
const num1 = Number('888');
const num2 = new Number('777');

console.log(num1 == 888); //statement 1
console.log(num2 == 777); // statement 2
```
````

code needs to be in code fences. If you have other code snippets please wrap them in backticks too

Author: Thanks. I have made the changes

#### --distractors--

Both statements are true

---

Both statements are false

---

Statement 1 is false while statement 2 is true

#### --answer--

> Statement 1 is true while statement 2 is false

Author: Hi gikf. thanks for the heads up. I made a mistake it should be the strick equality operator. I've already made the changes

### --question--

#### --text--

Which of the following is not an arithmetic operator in JavaScript

#### --distractors--

`%`

---

`+`

---

`--`

#### --answer--

`==`

### --question--

#### --text--

Which of the following will log the correct gross profit?

#### --distractors--

> `let store1_sales = 7500;
> let store2_sales = 6000;
> console.log("Total Sales: " +  store1_sales + store2_sales);`

Reviewer: ````suggestion
```js
let store1_sales = 7500;
let store2_sales = 6000;
console.log("Total Sales: " +  store1_sales + store2_sales);
```
````
a multiline code snippet requires three backticks above and the lang name, and three backticks below. Please fix the other code snippets

`Math.exp(base, power)`

---

`Math.raise(base, power)`

---

`Math.e(base, power)`

#### --answer--

`Math.pow(base, power)`

### --question--

#### --text--

Which of the following is the correct method for generating a random number in JavaScript?

#### --distractors--

`Math.rand(1,10)`

---

`Math.rand()`

---

`Math.random(1,10)`

#### --answer--

`Math.random()`

### --question--

#### --text--

Which of the following will return false when using isNaN()?

#### --distractors--

`isNaN("1998", "1999")`

---

`isNaN({})`

---

`isNaN(undefined)`

#### --answer--

`isNaN(null)`

### --question--

#### --text--

Which of the following will convert 172 to its hexadecimal equivalent?

#### --distractors--

`console.log((-0xa5).toString(2));`

---

`console.log((172).string(0x));`

---

`console.log((172).parseString(0x));`

#### --answer--

`console.log((172).toString(16));`

### --question--

#### --text--

Which of the following is the right way to convert string to integer?

#### --distractors--

`console.log(parseINT("300"))`

---

`console.log(int("300"))`

---

`console.log(integer("300"))`

#### --answer--

`console.log(parseInt("300"))`

### --question--

#### --text--

> Which of the following statement is false about parseFloat()?

Reviewer: ```suggestion
Which of the following statement is false about `parseFloat()`?
```

code inside a sentence also needs backticks, please fix the others

Author: Hi ilenia, i've made the changes

#### --text--

> What is the result of this code:

Reviewer 2: ```suggestion
What is the result of this code?
```

``` js
const num1 = Number('888');
const num2 = new Number('777');

> console.log(num1 === 888); //statement 1
> console.log(num2 === 777); // statement 2

Reviewer 2: ```suggestion
console.log(num1 === 888);
console.log(num2 === 777);
```

```

#### --distractors--

> Both statements are true

Reviewer 2: ````suggestion
```js
true
true
```
````

---

> Both statements are false

Reviewer 2: ````suggestion
```js
false
false
```
````

---

> Statement 1 is false while statement 2 is true

Reviewer 2: ````suggestion
```js
false
true
```
````

#### --answer--

> Statement 1 is true while statement 2 is false

Reviewer 2: ````suggestion
```js
true
false
```
````

### --question--

#### --text--

> Which of the following is not an arithmetic operator in JavaScript

Reviewer 2: ```suggestion
Which of the following is not an arithmetic operator in JavaScript?
```

#### --distractors--

`%`

---

`+`

---

`--`

#### --answer--

`==`

### --question--

#### --text--

Which of the following will log the correct gross profit?

#### --distractors--

``` js
let store1_sales = 7500;
let store2_sales = 6000;
console.log("Total Sales: " +  store1_sales + store2_sales);
```

---

``` js
let store1_sales = 7500;
let store2_sales = 6000;
console.log("Total Sales: " +  store1_sales - store2_sales);
```

---

``` js
let store1_sales = 7500;
let store2_sales = 6000;
console.log("Total Sales: " +  (store1_sales - store2_sales));
```

#### --answer--

``` js
let store1_sales = 7500;
let store2_sales = 6000;
console.log("Total Sales: " +  (store1_sales + store2_sales));
```

### --question--

#### --text--

> What is the result of this statement: `25 * 4 - 16 / (2 * 4) + 8`?

Reviewer 2: ```suggestion
What is the result of the statement `25 * 4 - 16 / (2 * 4) + 8`?
```

#### --distractors--

> 18.5

Reviewer 2: ```suggestion
`18.5`
```

---

> 107.5

Reviewer 2: ```suggestion
`107.5`
```

---

> 17

Reviewer 2: ```suggestion
`17`
```

#### --answer--

> 106

Reviewer 2: ```suggestion
`106`
```

### --question--

#### --text--

> Which operator to use when decrementing a variable in JavaScript?

Reviewer 2: ```suggestion
Which operator should you use when decrementing a variable in JavaScript?
```

#### --distractors--

`++`

---

`//`

---

`**`

#### --answer--

`--`

### --question--

#### --text--

Which of the following is not a bitwise operator?

#### --distractors--

`&`

---

`|`

---

`^`

#### --answer--

`>`

### --question--

#### --text--

Which of the following is the correct compound operator for getting the remainder?

#### --distractors--

`//=`

---

`\=`

---

`%%=`

#### --answer--

`%=`

### --question--

#### --text--

What is the difference between the `==` and `===` operators?

#### --distractors--

There is no difference. The comparison result will always have the same result.

---

> The `==` operator is used to check if the data type is the same while the `===` is used to check if the value is equal

Reviewer 2: ```suggestion
The `==` operator is used to check if the data type is the same while the `===` is used to check if the value is equal.
```

---

> The `==` operator can only be used with primitive types while the `===` operator is only used for objects

Reviewer 2: ```suggestion
The `==` operator can only be used with primitive types while the `===` operator is only used for objects.
```

#### --answer--

> The `==` operator only compares values whereas the `===` operator compares values and types

Reviewer 2: ```suggestion
The `==` operator only compares values whereas the `===` operator compares values and types.
```

> The `==` operator only compares values whereas `===`operator is more strict and will only return true if the value is equal and of the same data type

Reviewer 2: ```suggestion
The `==` operator only compares values whereas the `===` operator compares values and types
```

### --question--

#### --text--

Which of the following statements are false about unary plus (`+`) operators?

#### --distractors--

> It tries to convert the operand into a number

Reviewer 2: ```suggestion
It tries to convert the operand into a number.
```

---

> It is the fastest way to convert an operand to a number

Reviewer 2: ```suggestion
It is the fastest way to convert an operand to a number.
```

---

> It can convert operands with negative numbers except for hexadecimals

Reviewer 2: ```suggestion
It can convert operands with negative numbers except for hexadecimals.
```

#### --answer--

> It can't convert `true`, `false`, `null` into a number

Reviewer 2: ```suggestion
It can't convert `true`, `false`, or `null` into a number.
```

### --question--

#### --text--

> Which method in the Math standard built-in objects is used to round up a number?

Reviewer 2: ```suggestion
Which method in the built-in Math object is used to round up a number?
```

#### --distractors--

`Math.round(num)`

---

`Math.floor(num)`

---

`Math.fround(num)`

#### --answer--

`Math.ceil(num)`

### --question--

#### --text--

Which of the following Math object methods are used to raise a base to a power?

#### --distractors--

`Math.exp(base, power)`

---

`Math.raise(base, power)`

---

`Math.e(base, power)`

#### --answer--

`Math.pow(base, power)`

### --question--

#### --text--

Which of the following is the correct method for generating a random number in JavaScript?

#### --distractors--

`Math.rand(1,10)`

---

`Math.rand()`

---

`Math.random(1,10)`

#### --answer--

`Math.random()`

### --question--

#### --text--

> Which of the following will return false when using `isNaN()`?

Reviewer 2: ```suggestion
Which of the following will return `false`?
```

#### --distractors--

`isNaN("1998", "1999")`

---

`isNaN({})`

---

`isNaN(undefined)`

#### --answer--

`isNaN(null)`

### --question--

#### --text--

> Which of the following will convert 172 to its hexadecimal equivalent?

Reviewer 2: ```suggestion
Which of the following will convert `172` to its hexadecimal equivalent?
```

#### --distractors--

`(-0xa5).toString(2);`

---

`(172).string(0x);`

---

`(172).parseString(0x);`

#### --answer--

`(172).toString(16);`

### --question--

#### --text--

> Which of the following is the right way to convert string to integer?

Reviewer 2: ```suggestion
Which of the following is the right way to convert a string to an integer?
```

#### --distractors--

`parseINT("300")`

---

`int("300")`

---

`integer("300")`

#### --answer--

`parseInt("300")`

### --question--

#### --text--

> Which of the following statements is false about `parseFloat()`?

Reviewer 2: ```suggestion
Which of the following statements is false?
```

> Which of the following statement is false about `parseFloat()`?

Reviewer 2: ```suggestion
Which of the following statements is false about `parseFloat()`?
```

#### --distractors--

`parseFloat()` ignores trailing invalid characters

---

`parseFloat()` can parse a string that begins with `Infinity` or `-Infinity`

---

`parseFloat("0x1f")` will return 0

#### --answer--

`parseFloat()` requires two parameters

### --question--

#### --text--

> Which of the following is false about `Number.prototype.toFixed()`?

Reviewer 2: ```suggestion
Which of the following is false?
```

#### --distractors--

A number greater than 100 as an argument to the `Number.prototype.toFixed()` method will result to a range error

---

`Number.prototype.toFixed(numDecimalPlaces)` is used to return a number converted to a string with decimals places depending on the argument

---

Calling the `Number.prototype.toFixed()` method on a number object with decimal points and no argument will round the number to the nearest ones

#### --answer--

Using `Number.prototype.toFixed()` without an argument will default add a decimal point followed by a zero

### --question--

#### --text--

Which of the following is not a comparison operator?

#### --distractors--

`!=`

---

`>=`

---

`===`

#### --answer--

`!>`

### --question--

#### --text--

> Which of the following is false about operator precedence in JavaScriipt?

Reviewer 2: ```suggestion
Which of the following is false?
```

#### --distractors--

Exponentiation has a higher operator precedence than grouping

---

Postfix operators like `x++` or `x--` has higher precdence than prefix operators

---

Relational operators like `>` or `<` has higher precedence than equality operators

#### --answer--

Logical operator like `AND` and `OR` have higher precdence than bitwise operators

### --question--

#### --text--

> `Math.random()` returns a number randomly in what range?

Reviewer 2: ```suggestion
`Math.random()` returns a number in what range?
```

#### --distractors--

There is no difference. The comparison result will always have the same result.

---

The `==` operator is used to check if the data type is the same while the `===` is used to check if the value is equal

---

The `==` operator can only be used with primitive types while the `===` operator is only used for objects

#### --answer--

The `==` operator only compares values whereas `===`operator is more strict and will only return true if the value is equal and of the same data type

### --question--

#### --text--

Which of the following statements are false about unary plus (+) operators?

#### --distractors--

It tries to convert the operand into a number

---

It is the fastest way to convert an operand to a number

---

It can convert operands with negative numbers except for hexadecimals

#### --answer--

It can't convert `true`, `false`, `null` into a number

### --question--

#### --text--

Which method in the Math standard built-in objects is used to round up a number?

#### --distractors--

`Math.round(num)`

---

`Math.floor(num)`

---

`Math.fround(num)`

#### --answer--

`Math.ceil(num)`

### --question--

#### --text--

> Which of the following Math.object methods are used to raise a base to a power?

Reviewer 2: ```suggestion
Which of the following Math object methods are used to raise a base to a power?
```

#### --distractors--

`Math.exp(base, power)`

---

`Math.raise(base, power)`

---

`Math.e(base, power)`

#### --answer--

`Math.pow(base, power)`

### --question--

#### --text--

Which of the following is the correct method for generating a random number in JavaScript?

#### --distractors--

`Math.rand(1,10)`

---

`Math.rand()`

---

`Math.random(1,10)`

#### --answer--

`Math.random()`

### --question--

#### --text--

> Which of the following will return false when using isNaN()?

Reviewer 2: ```suggestion
Which of the following will return false when using `isNaN()`?
```

#### --distractors--

`isNaN("1998", "1999")`

---

`isNaN({})`

---

`isNaN(undefined)`

#### --answer--

`isNaN(null)`

### --question--

#### --text--

Which of the following will convert 172 to its hexadecimal equivalent?

#### --distractors--

> `console.log((-0xa5).toString(2));`

Reviewer 2: ```suggestion
`(-0xa5).toString(2);`
```

---

> `console.log((172).string(0x));`

Reviewer 2: ```suggestion
`(172).string(0x);`
```

---

> `console.log((172).parseString(0x));`

Reviewer 2: ```suggestion
`(172).parseString(0x);`
```

#### --answer--

> `console.log((172).toString(16));`

Reviewer 2: ```suggestion
`(172).toString(16);`
```

### --question--

#### --text--

Which of the following is the right way to convert string to integer?

#### --distractors--

> `console.log(parseINT("300"))`

Reviewer 2: ```suggestion
`parseINT("300")`
```

---

> `console.log(int("300"))`

Reviewer 2: ```suggestion
`int("300")`
```

---

> `console.log(integer("300"))`

Reviewer 2: ```suggestion
`integer("300")`
```

#### --answer--

> `console.log(parseInt("300"))`

Reviewer 2: ```suggestion
`parseInt("300")`
```

### --question--

#### --text--

Which of the following statement is false about `parseFloat()`?

#### --distractors--

`parseFloat()` ignores trailing invalid characters

---

> `parseFloat()` can parse a string that begins with "Infinity" or "-Infinity"

Reviewer 2: ```suggestion
`parseFloat()` can parse a string that begins with `Infinity` or `-Infinity`
```

#### --text--

What is the result of this code?

``` js
const num1 = Number('888');
const num2 = new Number('777');

console.log(num1 === 888);
console.log(num2 === 777);
```

#### --distractors--

```js
true
true
```

---

```js
false
false
```

---

```js
false
true
```

#### --answer--

```js
true
false
```

### --question--

#### --text--

Which of the following is not an arithmetic operator in JavaScript?

#### --distractors--

`%`

---

`+`

---

`--`

#### --answer--

`==`

### --question--

#### --text--

Which of the following will log the correct gross profit?

#### --distractors--

``` js
let store1_sales = 7500;
let store2_sales = 6000;

> console.log("Total Sales: " +  store1_sales + store2_sales);

Reviewer: ```suggestion
console.log("Total Sales: " + store1_sales + store2_sales);
```

```

---

``` js
let store1_sales = 7500;
let store2_sales = 6000;

> console.log("Total Sales: " +  store1_sales - store2_sales);

Reviewer: ```suggestion
console.log("Total Sales: " + store1_sales - store2_sales);
```

```

---

``` js
let store1_sales = 7500;
let store2_sales = 6000;

> console.log("Total Sales: " +  (store1_sales - store2_sales));

Reviewer: ```suggestion
console.log("Total Sales: " + (store1_sales - store2_sales));
```

```

#### --answer--

``` js
let store1_sales = 7500;
let store2_sales = 6000;

> console.log("Total Sales: " +  (store1_sales + store2_sales));

Reviewer: ```suggestion
console.log("Total Sales: " + (store1_sales + store2_sales));
```

`Math.pow(base, power)`

### --question--

#### --text--

Which of the following is the correct method for generating a random number in JavaScript?

#### --distractors--

`Math.rand(1,10)`

---

`Math.rand()`

---

`Math.random(1,10)`

#### --answer--

`Math.random()`

### --question--

#### --text--

Which of the following will return `false`?

#### --distractors--

`isNaN("1998", "1999")`

---

`isNaN({})`

---

`isNaN(undefined)`

#### --answer--

`isNaN(null)`

### --question--

#### --text--

Which of the following will convert `172` to its hexadecimal equivalent?

#### --distractors--

`(-0xa5).toString(2);`

---

`(172).string(0x);`

---

`(172).parseString(0x);`

#### --answer--

`(172).toString(16);`

### --question--

#### --text--

Which of the following is the right way to convert a string to an integer?

#### --distractors--

`parseINT("300")`

---

`int("300")`

---

`integer("300")`

#### --answer--

`parseInt("300")`

### --question--

#### --text--

Which of the following statements is false?

#### --distractors--

`parseFloat()` ignores trailing invalid characters

---

`parseFloat()` can parse a string that begins with `Infinity` or `-Infinity`

---

> `parseFloat("0x1f")` will return 0

Reviewer 3: ```suggestion
`parseFloat("0x1f")` will return `0`
```
Nitpick

#### --answer--

`parseFloat()` requires two parameters

### --question--

#### --text--

Which of the following is false?

#### --distractors--

A number greater than 100 as an argument to the `Number.prototype.toFixed()` method will result to a range error

---

`Number.prototype.toFixed(numDecimalPlaces)` is used to return a number converted to a string with decimals places depending on the argument

---

Calling the `Number.prototype.toFixed()` method on a number object with decimal points and no argument will round the number to the nearest ones

#### --answer--

> Using `Number.prototype.toFixed()` without an argument will default add a decimal point followed by a zero

Reviewer 3: ```suggestion
Using `Number.prototype.toFixed()` without an argument will by default add a decimal point followed by a zero
```
Nitpick

### --question--

#### --text--

Which of the following is not a comparison operator?

#### --distractors--

`!=`

---

`>=`

---

`===`

#### --answer--

`!>`

### --question--

#### --text--

Which of the following is false?

#### --distractors--

> Exponentiation has a higher operator precedence than grouping

Reviewer 3: ```suggestion
Exponentiation has a lower operator precedence than grouping
```
To be honest I initially wasn't entirely sure what _grouping_ meant in this context.

Author: i should have said grouping operator ()

Reviewer 3: This change got lost once again in the meantime. 😛

> Exponentiation has a higher operator precedence than the grouping operator `()`.

Reviewer 3: ```suggestion
Exponentiation has a lower operator precedence than the grouping operator `()`.
```
Grouping has the highest precedence, hasn't it?

Author: yes it does oops thanks
